src/frontends/onnx/frontend/src/op/quantize_linear.cpp

```diff
@@ -184,6 +195,36 @@ ov::OutputVector quantize_linear(ov::Output<ov::Node> x,
     const auto& y_scale_shape = y_scale.get_partial_shape();
     const auto& y_zero_point_shape = y_zero_point.get_partial_shape();
 
+    if (block_size > 1) {
+        if (x_shape.rank().is_static() && x_shape.rank().get_length() > 0 && x_shape[axis].is_static()) {
+            CHECK_VALID_NODE(node,
+                             x_shape[axis].get_length() % block_size == 0,
+                             "The input data axis size: ",
+                             x_shape[axis],
+                             " must be divisible by the block size: ",
+                             block_size);
+            ov::Shape target_shape(x_shape.rank().get_length() + 1, 1);
```

Comment:
Please, double check this part. I don't see any checks which allow us to be sure axis is definitely less than target_shape.size() - 2. I see only check inside normalize_axis where it verifies something like -r <= axis < r.